modules/afsocket/afsocket-source.c

```diff
@@ -212,6 +212,8 @@ afsocket_sd_kill_connection(AFSocketSourceConnection *connection)
 {
   log_pipe_deinit(&connection->super);
 
+  log_reader_close_connection(connection->reader);
```

Comment:
I think this could be applied to other source plugins.

Reply:
Right, I'll check them. The name "close_connection" is also wrong here. LogReader has nothing to do with connections. "close_transport" might be a bit better name.